apps/backend/src/services/audit_service.py

```
@@ -0,0 +1,443 @@
+                generated_at=datetime.now(timezone.utc),
+                overall_metrics=overall_metrics,
+                by_data_type=by_data_type,
+                by_collection_method=by_collection_method,
+                by_hour=by_hour,
+                performance_trends=performance_trends,
+                recommendations=recommendations,
+            )
+
+        except Exception as e:
+            logger.error(f"Error generating performance report: {e}")
+            raise DatabaseOperationError(f"Failed to generate performance report: {str(e)}", "get_performance_report")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)